### ISSUE TITLE
An empty checkpoint folder without memory.pkl crashes training

### ISSUE THREAD

**Issue by mdnunez** [stefanradev93/BayesFlow#64]
Hey everyone,

Great work on the new version of the package. The notebook examples are quite clear and easy to adapt. 

In case you didn't know about this error, if you create an empty checkpoint folder ahead of running `bayesflow.trainers.Trainer() `this seems to crash the function. You can let `Trainer()` create the folder for you, but this may be a bit annoying for some analysis pipelines. For example, this will crash for me when using the `master` branch:

```
os.makedirs('checkpoint_model1')
trainer = bf.trainers.Trainer(
    amortizer=amortizer, 
    generative_model=generative_model, 
    configurator=configurator,
    checkpoint_path='checkpoint_model1')
```

Cheers,

**Comment by stefanradev93** [comment]
Hi Michael, 

thanks for pointing this out! The `Trainer` was actually intended to create a new folder for each amortizer in order to keep trained instances separated and not overwrite anything. What I can do is include a check for the specific default filenames used for memory, loss history, and networks in an existing folder, and only accept it if there are no filename conflicts.

**Comment by leolo23** [comment]
Hi Stefan,
I found myself having the same problem. I think what is causing it is line 747 in "helper_classes.py", it is checking whether the folder exists instead of weather the memory file exist: `if os.path.exists(file_path):`, instead I think it should be `if os.path.exists(memory_path):`

**Comment by stefanradev93** [comment]
The issue should be resolved in version 1.1.2. Thanks!
